### PR TITLE
Lammps: Fix final structure not being stored

### LINKED ISSUE #1340: Changes to lammps output hdf5
The format of lammps h5 output seems to have changed recently. This broke the pyrionFileReader of ovito https://github.com/ovito-org/pyironFileReader.
I implemented a fix in https://github.com/ovito-org/pyironFileReader/pull/1, which reads pbc and species from input/structure instead of
the removed output/structure. This works for both, old and new lammps hdf5 files. However, I am not sure whether it could cause problems due to internal mappings changing species or similar.
So the question is has output/structure been removed because it  doubled data or are there actual differences?

### PR DESCRIPTION
**Pull request by jan-janssen**
I guess this was broken in 0.4.2 - https://github.com/pyiron/pyiron_atomistics/releases/tag/pyiron_atomistics-0.4.2

### CODE REVIEW

File: pyiron_atomistics/lammps/base.py [hunk 1]
```diff
@@ -459,9 +459,11 @@ def collect_output(self):
...
         )
-
-        if len(self.output.cells) > 0:
-            final_structure = self.get_structure(iteration_step=-1)
+        if len(hdf_dict["output/generic/cells"]) > 0:
+            final_structure = self.structure.copy()
+            final_structure.indices = hdf_dict["output/generic/indices"][-1]
+            final_structure.positions = hdf_dict["output/generic/positions"][-1]
+            final_structure.cell = hdf_dict["output/generic/cells"][-1]
             if final_structure is not None:
```
Comment (pmrv) [inline pyiron_atomistics/lammps/base.py]: Can never be `None` now, so if check is unnecessary.

File: pyiron_atomistics/lammps/base.py [hunk 2]
```diff
@@ -459,9 +459,11 @@ def collect_output(self):
             ),
             group_name="output",
         )
-
-        if len(self.output.cells) > 0:
-            final_structure = self.get_structure(iteration_step=-1)
```
Comment (pmrv) [inline pyiron_atomistics/lammps/base.py:464]: Does that mean the `output.cells` and `get_structure` don't work anymore as well or is this somehow restored by writing the final structure below?

Comment (jan-janssen) [reply pyiron_atomistics/lammps/base.py:464]: It does not work until the output is stored in the HDF5 file, which only happens at the end of `collect_output()`.

**Review by Leimeroth — APPROVED** [review]

File: pyiron_atomistics/lammps/base.py (hunk 1 above)
Comment (jan-janssen) [reply pyiron_atomistics/lammps/base.py]: I removed the if statement

**Pull request merged**